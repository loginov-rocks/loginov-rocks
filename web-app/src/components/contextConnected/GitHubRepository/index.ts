import { GitHubRepository } from './GitHubRepository';

export { GitHubRepository };
