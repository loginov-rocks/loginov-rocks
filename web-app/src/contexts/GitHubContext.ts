import { GitHubData } from '@loginov-rocks/loginov-rocks-shared';
import { createContext, useContext } from 'react';

export const GitHubContext = createContext<GitHubData | null>(null);

export const useGitHubContext = (): GitHubData | null => useContext(GitHubContext);
