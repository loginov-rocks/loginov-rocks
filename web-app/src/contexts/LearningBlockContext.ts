import { createContext, useContext } from 'react';

interface ContextValue {
  addTimeToComplete: (timeToComplete: number) => void;
}

export const LearningBlockContext = createContext<ContextValue>({
  addTimeToComplete: () => undefined,
});

export const useLearningBlockContext = (): ContextValue => useContext(LearningBlockContext);
