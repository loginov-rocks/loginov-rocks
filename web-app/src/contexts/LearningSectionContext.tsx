import { createContext, useContext } from 'react';

interface ContextValue {
  addCompleted: (completed: string) => void;
  filterYear: string | null;
}

export const LearningSectionContext = createContext<ContextValue>({
  addCompleted: () => undefined,
  filterYear: null,
});

export const useLearningSectionContext = (): ContextValue => useContext(LearningSectionContext);
