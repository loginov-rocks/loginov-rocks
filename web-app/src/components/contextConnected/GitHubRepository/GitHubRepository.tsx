import * as React from 'react';

import { GitHubRepositoryDetails } from 'components/shared/GitHubRepositoryDetails';
import { ExternalLink } from 'components/shared/ExternalLink';
import { useGitHubContext } from 'contexts/GitHubContext';

interface Props {
  gitHubId: string;
}

export const GitHubRepository: React.FC<Props> = ({ gitHubId }) => {
  const gitHubData = useGitHubContext();
  const repository = gitHubData && gitHubData.repos
    ? gitHubData.repos.find((gitHubRepo) => gitHubRepo.name === gitHubId) : undefined;

  let title = <strong>{gitHubId}</strong>;

  if (repository && repository.homepageUrl) {
    title = <ExternalLink href={repository.homepageUrl} label="Link to Project Homepage">{title}</ExternalLink>;
  }

  if (repository && repository.isArchived) {
    title = <s title="Archived">{title}</s>;
  }

  return (
    <>

      {title}

      {repository && (
        <>
          {' '}
          /
          {' '}
          <ExternalLink href={repository.url} label="Link to Repository">GitHub</ExternalLink>
        </>
      )}

      {repository && (
        <>
          <br />
          <GitHubRepositoryDetails repository={repository} />
        </>
      )}

    </>
  );
};
