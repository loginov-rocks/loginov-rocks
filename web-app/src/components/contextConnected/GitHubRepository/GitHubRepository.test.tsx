/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import * as GitHubContext from 'contexts/GitHubContext';

import { GitHubRepository } from './GitHubRepository';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const gitHubDataMock = require('../../../../../shared/src/GitHub/__fixtures__/gitHubData.json');

jest.spyOn(GitHubContext, 'useGitHubContext')
  .mockImplementation(() => gitHubDataMock);

it('matches snapshot without GitHub details', () => {
  const wrapper = shallow(<GitHubRepository gitHubId="Some work" />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot with GitHub details', () => {
  const wrapper = shallow(<GitHubRepository gitHubId="Angular-Gulp-Boilerplate" />);

  expect(wrapper).toMatchSnapshot();
});
