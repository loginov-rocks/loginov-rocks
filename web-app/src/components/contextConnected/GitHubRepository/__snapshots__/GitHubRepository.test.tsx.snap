// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with GitHub details 1`] = `
<Fragment>
  <ExternalLink
    href="https://www.npmjs.com/package/angular-gulp-boilerplate"
  >
    <strong>
      Angular-Gulp-Boilerplate
    </strong>
  </ExternalLink>
   
  /
   
  <ExternalLink
    href="https://github.com/loginov-rocks/Angular-Gulp-Boilerplate"
  >
    GitHub
  </ExternalLink>
  <br />
  <GitHubRepositoryDetails
    repository={
      Object {
        "description": "Clean but full featured AngularJS boilerplate using Gulp workflow and best practices",
        "homepageUrl": "https://www.npmjs.com/package/angular-gulp-boilerplate",
        "isArchived": false,
        "language": "JavaScript",
        "latestTag": "v1.0.10",
        "name": "Angular-Gulp-Boilerplate",
        "stars": 29,
        "updatedAt": 1614276648000,
        "url": "https://github.com/loginov-rocks/Angular-Gulp-Boilerplate",
      }
    }
  />
</Fragment>
`;

exports[`matches snapshot without GitHub details 1`] = `
<Fragment>
  <strong>
    Some work
  </strong>
</Fragment>
`;
